### ISSUE TITLE
Refactor NavigationUiComponent

### ISSUE THREAD

**Issue by kacperknapik** [SAP/spartacus#7206]
### Needs fixing:
- [x] a11y-related logic

**Comment by kacperknapik** [comment]
**QA steps**
- log in
- 'My Account' should be accessible when tabbing
- bold category names in menu dropdowns (eg. CAMERAS and ACCESSORIES in BRANDS) should not be accessible when tabbing

**Comment by KateChuen** [comment]
Tested on dev10 on chrome, firefox & safari 
✅'My Account' should be accessible when tabbing
✅bold category names in menu dropdowns (eg. CAMERAS and ACCESSORIES in BRANDS) should not be accessible when tabbing
